src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModel.java

```diff
@@ -0,0 +1,10 @@
+package fi.aalto.cs.apluscourses.presentation;
+
+import fi.aalto.cs.apluscourses.utils.ObservableProperty;
+import org.jetbrains.annotations.NotNull;
+
+public class MainViewModel {
+
+  @NotNull
+  public final ObservableProperty<CourseViewModel> courseViewModel = new ObservableProperty<>(null);
```

Comment:
Nit (and not urgent to decide on for this pr): we could decide on some consistent way to choose between `public final` and getters (for an example, our exception classes have private finals with getters)

Reply:
What makes (perhaps) the difference here is the nature of `ObservableProperty<>`. Consider:
```java
mainViewModel.getCourseViewModel().set(new CourseViewModel(...));
```
Here we want to *set* the course view model so calling *getter* for that is a bit confusing. That's why I thought
```java
mainViewModel.courseViewModel.set(new CourseViewModel(...));
```
is better although it is not that good OOP.

Reply:
Good point, the `get...set` is indeed somewhat confusing :smile: